File: SWP.Core/Entities/Warranty.cs
using SWP.Core.GMSAttribute;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SWP.Core.Entities
{
    [GMSTableName("warranty")]
    public class Warranty
    {
        [Column("warranty_id")]
        [Key]
        public int WarrantyId { get; set; }

        [Column("service_ticket_detail_id")]
        [Required]
        public int ServiceTicketDetailId { get; set; }

        [Column("start_date")]
        [Required]
        public DateTime StartDate { get; set; }

        [Column("end_date")]
        [Required]
        public DateTime EndDate { get; set; }

        [Column("status")]
        public byte? Status { get; set; }

        [Column("is_deleted")]
        public int IsDeleted { get; set; } = 0;

        // 
        public ServiceTicketDetail ServiceTicketDetail { get; set; } = null!;
    }
}

File: GarageProject-BE-Springboot-MainFlow/src/main/java/fptu/edu/vn/training/entity/Warranty.java
package fptu.edu.vn.training.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "warranty")
public class Warranty {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "warranty_id")
    private Integer warrantyId;

    @Column(name = "service_ticket_detail_id", nullable = false)
    private Integer serviceTicketDetailId;

    @Column(name = "start_date", nullable = false)
    private LocalDateTime startDate;

    @Column(name = "end_date", nullable = false)
    private LocalDateTime endDate;

    @Column(name = "status")
    private Byte status;

    @Column(name = "is_deleted", nullable = false)
    @Builder.Default
    private Integer isDeleted = 0;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "service_ticket_detail_id", insertable = false, updatable = false)
    private ServiceTicketDetail serviceTicketDetail;
}
